src/middleware/validate_zome_call_payload_size.rs

```diff
@@ -0,0 +1,53 @@
+use axum::{
+    extract::{Query, Request, State},
+    http::StatusCode,
+    middleware::Next,
+    response::{IntoResponse, Response},
+    Json,
+};
+
+use crate::{http::HcGwErrorResponse, routes::PayloadQuery, service::AppState};
+
+#[tracing::instrument(skip(state, request, next))]
+pub async fn validate_zome_call_payload_size(
```

Comment:
I'm still a little torn about this as middleware. Middleware is supposed to be re-usable. So checking authentication or some other request headers, or pre-processing request content, or handling upgrading requests etc. 

At the moment we don't have requirements for any other requests that need to check a payload size. Even if we do, this is tied to a configuration setting so it's not necessarily re-usable.

I'm not asking for this to be changed right now, but I think we could save some complexity by merging it into the GET request handler.

Reply:
I see, I have moved the payload checks to the route for simplicity